pkg/certificate/mrc_reconciler.go

```diff
@@ -1,214 +1,228 @@
+	if desiredSigningMRC == nil || desiredValidatingMRC == nil {
+		log.Error().Err(ErrUnexpectedNilMRC).Msg("unexpected nil MRC provided when getting cert issuers from MRCs")
+		return false, ErrUnexpectedNilMRC
+	}
+
+	m.mu.Lock()
+	signingIssuer := m.signingIssuer
+	validatingIssuer := m.validatingIssuer
+	m.mu.Unlock()
+
+	// if the issuers are already set to the desired value, return
+	if signingIssuer != nil && signingIssuer.ID == desiredSigningMRC.Name && validatingIssuer != nil && validatingIssuer.ID == desiredValidatingMRC.Name {
+		log.Debug().Msgf("issuers already set to the desired value. Will not update issuers: validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+		return false, nil
+	}
+
+	// if desiredSigningMRC != desiredValidatingMRC and both MRCs have active intents, their state is non
+	// deterministic. To avoid continuously resetting the issuers, only update
+	// if the issuers have not already been set
+	if desiredSigningMRC != desiredValidatingMRC && desiredSigningMRC.Spec.Intent == v1alpha2.ActiveIntent &&
+		desiredValidatingMRC.Spec.Intent == v1alpha2.ActiveIntent && signingIssuer != nil && validatingIssuer != nil {
+		log.Debug().Msgf("issuers already set and MRC intents are non deterministic; validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+
+		// if the issuers match either desired MRC, don't update. Note: the issuers are not set to their
+		// desired values, but are in an acceptable state.
+		if (desiredSigningMRC.Name == signingIssuer.ID || desiredSigningMRC.Name == validatingIssuer.ID) &&
+			(desiredValidatingMRC.Name == signingIssuer.ID || desiredValidatingMRC.Name == validatingIssuer.ID) {
+			log.Debug().Msgf("Will not update issuers to avoid repeated updates; validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+			return false, nil
 		}
+		// update since the issuers do not match the desired MRCs
+		return true, nil
 	}
 
-	return m.mrcClient.UpdateMeshRootCertificate(mrc)
+	return true, nil
 }
 
-// isTerminal checks if the MRC is in a terminal state.
-func isTerminal(mrc *v1alpha2.MeshRootCertificate) bool {
-	// TODO(5046): check if the mrc is in a terminal state.
-	return true
+func (m *Manager) updateIssuers(signingIssuer, validatingIssuer *issuer) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.signingIssuer = signingIssuer
+	m.validatingIssuer = validatingIssuer
+	log.Trace().Msgf("setting issuers for validating[%s] and signing[%s]", validatingIssuer.ID, signingIssuer.ID)
+	return nil
 }
 
-// state and condition get updated together.
-func (m *Manager) updateMRCState(mrc *v1alpha2.MeshRootCertificate) error {
-	// TODO(5046): update the MRC state & conditions, and issue the update via the MRCClient.
+func (m *Manager) getCertIssuers(desiredSigningMRC, desiredValidatingMRC *v1alpha2.MeshRootCertificate) (*issuer, *issuer, error) {
+	if desiredSigningMRC == nil || desiredValidatingMRC == nil {
```

Comment:
There's a lot of duplicative nil checks. These are internal methods, so no need to be as defensive about nil calls, which should be applied at the API level (exported funcs)